### DISCUSSION TITLE
Resultant of Passing a Vector and the _grad Operator

### DISCUSSION THREAD

**Discussion by jsevic** [idaholab/moose#26117] [Q&A General] [answered]
Hello,

If I pass a row vector in a custom AuxKernek using this command:

`   _direction(getParam<RealVectorValue>("direction"))
`
and then return

` _direction * _grad_coupled_n[_qp] 
`
will this be a scalar?

Another related question, in the AuxKernel if the row vector above is called row_vector, is this passed as direction = '[1 0 0]' for the x-direction?

Thank you!

**Comment by GiudGiud** [comment, marked as answer]
Hello

What is the type of `_grad_coupled_n` ? This is from your kernel right? I have not seen it in the moose base clases.

If it s a RealVectorValue / a VariableGradient then yes. * is the dot-product for 'vectors' (of size 2/3, RealVectorValue)

If you want to specify a vector in the input, 1 0 0 would be the X-direction indeed, you dont need to try to pass it vertically like 
```
row_vector = '1
                         0
                         0'
```
 (not like above)

**Comment by jsevic** [comment]
OK this works thanks, `_grad_coupled_n` should have been pasted as  `_grad_coupled_V` and what really helped too was I found this AuxKernel VectorVariableMagnitudeAux and now my MOOSE test cases match my hand calculations for integrating current density to obtain current.
